### PR TITLE
Named catalogs

### LINKED ISSUE #41: Named catalogs
When a file is uploaded, a name can optionally be specified. The file should be capable of being retrieved and stored (and overwritten) using this name.

### PR DESCRIPTION
**Pull request by cryptoquick**
- [x] Store named catalog
- [x] Retrieve named catalog
- [x] File name encoding

Closes #41

### CODE REVIEW

**Comment by cryptoquick** [comment]
To test:
`POST http://localhost:7000/store_named/<your pk>/<your name>`
`GET http://localhost:7000/retrieve_named/<your pk>/<your name>`

**Comment by cryptoquick** [comment]
Oh shoot, just realized this could be exploited to read just about any user file, but that's okay, I can fix that quickly

File: src/backend/fs.rs [hunk 1]
```diff
@@ -138,15 +143,16 @@ pub async fn write_catalog(
 
     let write_pk_str = write_pk_str.to_owned();
     let file_hash = file_hash.to_string();
+    let name = name.unwrap_or(file_hash);
```
Comment (rustchain64) [inline src/backend/fs.rs:146]: so unwrap_or, will use file_hash if name is_none()
That is what I think here. 
This is what you discussed giving the user the option to name a file.

**Review by rustchain64 — APPROVED** [review]

**Comment by cryptoquick** [comment]
Okay, in my experimentation, that probably couldn't have happened since the axum router would consider most exploitable routes gobbledygook, but just to be safe, I added the percent-encoding check anyway.

File: src/backend/fs.rs (hunk 1 above)
Comment (cryptoquick) [reply src/backend/fs.rs:146]: Correct.

**Pull request merged**